### DISCUSSION TITLE
How to establish a connection with the client only, without establishing any connection with the server.

### DISCUSSION THREAD

**Discussion by zekai-li** [mitmproxy/mitmproxy#7300] [Q&A]
I only need to intercept the client's files, modify them, and then return them to the client without establishing any connection with the server. This way, the implanted program will not be detected by the server at all. I've noticed that in some environments, the TLS handshake with the server fails, and I haven't found the reason yet.

**Comment by zekai-li** [comment]
<img width="715" alt="image" src="https://github.com/user-attachments/assets/b427ad83-c02d-4a22-8846-3c0d8229a503">

**Comment by zekai-li** [comment]
<img width="893" alt="image" src="https://github.com/user-attachments/assets/d9f93c1f-98ff-409d-baa8-dd731a5d8967">

**Comment by mhils** [comment]
This sounds like you want to set [`connection_strategy`](https://docs.mitmproxy.org/stable/concepts-options/#connection_strategy) to lazy. :)

**Reply by zekai-li** [reply]
<img width="376" alt="image" src="https://github.com/user-attachments/assets/ef9354e5-4800-41f2-982b-1d214739d2c9">
There is a peculiar phenomenon: when I use a remote server as a proxy, some websites cannot be accessed, and it seems that some SSL processes fail. However, using naiveproxy works. You mentioned that the lazy method also doesn't work.

**Reply by zekai-li** [reply]
<img width="947" alt="image" src="https://github.com/user-attachments/assets/43e66528-4619-4b0a-bfc7-807f5749aa53">
use lazy connection strategy
